cmd/ingester/app/consumer/consumer.go

```diff
@@ -75,8 +75,10 @@ func New(params Params) (*Consumer, error) {
 // Start begins consuming messages in a go routine
 func (c *Consumer) Start() {
 	c.deadlockDetector.start()
+	c.doneWg.Add(1)
 	go func() {
 		c.logger.Info("Starting main loop")
+		c.doneWg.Done()
```

Comment:
```suggestion
		defer c.doneWg.Done()
		c.logger.Info("Starting main loop")
```

Reply:
Doing, that cause test to timeout because this [channel](https://github.com/jaegertracing/jaeger/blob/main/cmd/ingester/app/consumer/consumer.go#L80) is not Closed at all for some reason. I investigate it further and it seems like this [method ](https://github.com/jaegertracing/jaeger/blob/main/cmd/ingester/app/consumer/consumer_test.go#L81)is not closing the partition channel properly at least. The test that causes the code to timeout is https://github.com/jaegertracing/jaeger/blob/main/cmd/ingester/app/consumer/consumer_test.go#L122. It seems like the bug is happening on the "github.com/Shopify/sarama/mocks" side.

Reply:
I take it back. It is happening on the Jaeger Side. I will submit a correct push shortly